src/Microsoft.CodeQuality.Analyzers/UnitTests/Maintainability/RemoveUnusedLocalsTests.Fixer.cs

```diff
@@ -30,5 +32,484 @@ protected override CodeFixProvider GetCSharpCodeFixProvider()
+class C
+{
+    int M()
+    {
+        int b = 0;
+        b = 0;
+        return b;
+    }
+}
+";
+            VerifyCSharpFix(code, fix, allowNewCompilerDiagnostics: true);
+            VerifyCSharpFixAll(code, fix, allowNewCompilerDiagnostics: true);
+        }
+
+        [Fact]
+        public void UnusedLocal_JointAssignment_RemoveSecond_CSharp()
+        {
+            var code = @"
+class C
+{
+    int M()
+    {
+        int a = 0;
+        int b = 0;
+        a = b = 0;
+        return a;
+    }
+}
+";
+            var fix = @"
+class C
+{
+    int M()
+    {
+        int a = 0;
+        a = 0;
+        return a;
+    }
+}
+";
+            VerifyCSharpFix(code, fix, allowNewCompilerDiagnostics: true);
+            VerifyCSharpFixAll(code, fix, allowNewCompilerDiagnostics: true);
+        }
+
+        [Fact]
+        public void UnusedLocal_JointAssignment_RemoveBoth_CSharp()
+        {
+            var code = @"
+class C
+{
+    int M()
+    {
+        int a = 0;
+        int b = 0;
+        a = b = 0;
+        return 0;
+    }
+}
+";
+            var fix = @"
+class C
+{
+    int M()
+    {
+        return 0;
+    }
+}
+";
+            // fixAll removes unused locals found in a single iteration.
+            // a is used in the first iteration.
+            var fixAll = @"
+class C
+{
+    int M()
+    {
+        int a = 0;
+        a = 0;
+        return 0;
+    }
+}
+";
+            VerifyCSharpFix(code, fix, allowNewCompilerDiagnostics: true);
+            VerifyCSharpFixAll(code, fixAll, allowNewCompilerDiagnostics: true);
+        }
+
+        [Fact]
+        public void UnusedLocal_LocalFunction_CSharp()
+        {
+            var code = @"
+class C
+{
+    int M()
+    {
+        int L() { int a = 1; return 1; }
+        return 1;
+    }
+}
+";
+            var fix = @"
+class C
+{
+    int M()
+    {
+        return 1;
+    }
+}
+";
+            VerifyCSharpFix(code, fix, allowNewCompilerDiagnostics: true);
+            VerifyCSharpFixAll(code, fix, allowNewCompilerDiagnostics: true);
+        }
+
+        [Fact (Skip = "https://github.com/dotnet/roslyn/issues/22921")]
+        public void UnusedLocal_Lambda_CSharp()
+        {
+            var code = @"
+class C
+{
+    int M()
+    {
+        Func<int> lambda = () =>
+        {
+            int a = 4;
+            Func<int> internalLambda = () => { int bbb = 4; return 2; };
+            return 1;
+        };
+        return 1;
+    }
+}
+";
+            var fix = @"
+class C
+{
+    int M()
+    {
+        return 1;
+    }
+}
+";
+            VerifyCSharpFix(code, fix, allowNewCompilerDiagnostics: true);
+            VerifyCSharpFixAll(code, fix, allowNewCompilerDiagnostics: true);
+        }
+
+        [Fact]
+        public void UnusedLocal_BaseFunctionality_Basic()
+        {
+            var code = @"
+Public Class C
+    Function F() As Integer
+        Dim a As Integer = 0 ' inline comment also to be deleted. 
+        Dim b As Integer
+        b = 0
```

Comment:
Again, I would recommend adding a test where right hand side here is a method call.

Reply:
Please see _Parameters_ in this iteration or _ParametersAndMethodCalls_ in the next one.

---
In reply to: [152383098](https://github.com/dotnet/roslyn-analyzers/pull/1336#discussion_r152383098) [](ancestors = 152383098)

Reply:
@ivanbasov That test only covers method calls in initializers at declaration. I meant method calls in subsequent assignment statements. Regardless, I confirmed your claim and in both cases the variable is considered as used.

Reply:
@ivanbasov It seems compiler generates this diagnostic only if the locals are assigned constant values at all def sites. Is this correct?